lib/identity_cache/should_use_cache.rb

```diff
@@ -0,0 +1,13 @@
+module IdentityCache
+  module ShouldUseCache
+    extend ActiveSupport::Concern
+
+    included do
+      unless self.respond_to?(:should_use_cache?)
+        def self.should_use_cache?
```

Comment:
That wouldn't be a problem if AssociatedRecord defined should_use_cache? directly like that, but I can see how that would be a problem if it was defined from a mixin.  However, I still don't like you you are dynamically defining should_use_cache? on the class itself here.

Instead, you could change

```ruby
child_class.send(:include, ShouldUseCache) unless child_class.include?(ShouldUseCache)
```

to

```
child_class.send(:include, ShouldUseCache) unless child_class.respond_to?(:should_use_cache?)
```

in add_parent_expiry_hook.